qiskit_optimization/converters/linear_inequality_to_penalty.py

```diff
@@ -0,0 +1,331 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Converter to convert a problem with inequality constraints to unconstrained with penalty terms."""
+
+import itertools
+import logging
+from math import fsum
+from typing import Optional, cast, Union, Tuple, List
+
+import numpy as np
+
+from .quadratic_program_converter import QuadraticProgramConverter
+from ..exceptions import QiskitOptimizationError
+from ..problems.constraint import Constraint, ConstraintSense
+from ..problems.quadratic_objective import QuadraticObjective
+from ..problems.quadratic_program import QuadraticProgram
+from ..problems.variable import Variable
+
+logger = logging.getLogger(__name__)
+
+
+class LinearInequalityToPenalty(QuadraticProgramConverter):
+    r"""Convert a problem of inequality constraints to unconstrained with penalty terms.
+
+    There are some constraints which do not require to add slack variables to
+    construct penalty terms [1].Supported inequality constraint in this class is shown below.
+
+    .. math::
+
+        \begin{array}{}
+        \text { Inequality Constraint } & & \text { Equivalent Penalty } \\
+        x+y \leq 1 & \rightarrow & P(x y) \\
+        x+y \geq 1 & \rightarrow & P(1-x-y+x y) \\
+        x \leq y & \rightarrow  & P(x-x y) \\
+        x \geq y & \rightarrow  & P(y-x y) \\
+        x+y + z \leq 1 & \rightarrow & P\left(x y + y z + z x \right) \\
```

Comment:
This case is included in `\sum_i^n x_i \leq 1`.
`x + y <= 1` and `x + y >= 1` are also part of the last two cases, but it is good to keep them for ease of understanding. 
```suggestion
```